### PR TITLE
Less asynchrony for Binary::getContent

### PR DESCRIPTION
**Pull request by ajs6f**
This changeset removes one level of asynchrony from the path between binary data and HTTP by switching the signatures of `Binary::getContent` from `CompletionStage<InputStream>` to `InputStream`. This is appropriate because the only use `trellis-http` makes use of the `CompletionStage<InputStream>` is to immediately call `toCompletableFuture().join()` on it, which is absurd. We should rely going forward on the practice we've developed of answering HTTP requests using `@Suspended` by getting a `CompletionStage` from one or more of `ResourceService`, `MementoService`, `BinaryService`, etc.. The types _inside_ those `CompletionStage`s would themselves feature methods that return ordinary (non-promise/future) types. This changeset causes that to be the case.

### CODE REVIEW

File: core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java [hunk 1]
```diff
@@ -355,29 +356,22 @@ private ResponseBuilder getLdpNr(final ResponseBuilder builder) {
...
-                try (final InputStream binary = getBinaryStream(dsid, getRequest()).toCompletableFuture().join()) {
-                    IOUtils.copy(binary, out);
-                }
-            }
-        };
-
-        return builder.entity(stream);
+        // TODO -- with JDK 9 use InputStream::transferTo instead of IOUtils::copy
+        return getBinaryStream(dsid, getRequest())
+                        .thenApply(in -> (StreamingOutput) out -> IOUtils.copy(in, out))
```
Comment (acoburn) [inline core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java]: Should the input stream be closed here?

File: core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java [hunk 2]
```diff
@@ -355,29 +356,22 @@ private ResponseBuilder getLdpNr(final ResponseBuilder builder) {
...
-            public void write(final OutputStream out) throws IOException {
-                // TODO -- with JDK 9 use InputStream::transferTo instead of IOUtils::copy
-                try (final InputStream binary = getBinaryStream(dsid, getRequest()).toCompletableFuture().join()) {
-                    IOUtils.copy(binary, out);
-                }
-            }
-        };
-
-        return builder.entity(stream);
+        // TODO -- with JDK 9 use InputStream::transferTo instead of IOUtils::copy
```
Comment (acoburn) [inline core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java]: This TODO item can be removed. It is just cluttering things up, and I've been meaning to take it out. (Though feel free to leave it in, if you prefer)

Comment (ajs6f) [reply core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java]: Do you think this is invalid? I thought it was still useful…

Comment (acoburn) [reply core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java]: Sure, we can keep it here as a reminder

File: core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java (hunk 1 above)
Comment (ajs6f) [reply core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java]: Hm, I think you might be right. I'll take a look.

Comment (ajs6f) [reply core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java]: Good catch. I've fixed it.

**Pull request merged**